quadtree.js

```diff
@@ -13,11 +13,12 @@ class Point {
 }
 
 class Rectangle {
-  constructor(x, y, w, h) {
+  constructor(x, y, w, h, data) {
     this.x = x;
     this.y = y;
     this.w = w;
     this.h = h;
+    this.userData = data;
```

Comment:
This change may be useful in the future but it doesn't seem related to the delete and move functionality of this particular pull request. Can we take it out of this branch for now?